### PR TITLE
Add AzureFirewall swagger and examples

### PR DESCRIPTION
**Pull request by joannakleong**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by sergey-shandar** [comment]
@ravbhatnagar new spec AzureFirewall.

**Comment by joannakleong** [comment]
Hi @ravbhatnagar , @sergey-shandar , FYI we had gotten ARM approval when we prepared for private preview (https://github.com/Azure/azure-rest-api-specs-pr/pull/246)

File: specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/azureFirewall.json [hunk 1]
```diff
@@ -0,0 +1,648 @@
...
+            }
+         }
+      }
+   },
+   "definitions":{  
+      "AzureFirewallIPConfigurationPropertiesFormat":{  
+         "properties":{  
+			"privateIPAddress": {
+			  "type": "string",
+			  "description": "PrivateIPAddress of the network interface IP Configuration. This field is populated in the output."
```
Comment (MadhusudhanRavi) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/azureFirewall.json]: This is actually the front end IP address of the Internal Loadbalancer. we can check with Yair on how to expose this and if we should say this is the IPAddress they need to set as nexthop in UDR.

**Comment by dsgouda** [comment]
@MikhailTryakhov 
Can you take a look at this PR. Is this orthogonal to the changes [here](https://github.com/Azure/azure-rest-api-specs/pull/3261)

**Review by MikhailTryakhov — APPROVED** [review]
LGTM. It's a new feature, Samer approved to push it to master directly.
Please merge Network-2018-06-01 first and this one after

File: specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/azureFirewall.json [hunk 2]
```diff
@@ -0,0 +1,648 @@
...
+               "description":"The action type of a rule collection"
+            },
+            "rules":{  
+               "type":"array",
+               "items":{  
+                  "$ref":"#/definitions/AzureFirewallNetworkRule"
+               },
+               "description":"Collection of rules used by a network rule collection."
+            },
+            "provisioningState":{
```
Comment (MadhusudhanRavi) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/azureFirewall.json]: We only care about provisioning state on AzureFirewallProperties why do we have to mentioned provisioning state on these child resources? we don't have operations on them. Also the same is for Ipconfig as well.

Comment (MadhusudhanRavi) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/azureFirewall.json]: I see that other resources have it and NRP returns this value. So you can ignore this comment.

**Review by sergey-shandar — APPROVED** [review]

**Comment by sergey-shandar** [comment]
@joannakleong follow @MikhailTryakhov advise and then let me know when it can be merged.

**Comment by dsgouda** [comment]
@sergey-shandar This is good to merge

**Comment by joannakleong** [comment]
Thanks everyone for the prompt review! @sergey-shandar Before merging, I'd like to resolve @MadhusudhanRavi comments on the description fields for a couple properties

File: specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/azureFirewall.json (hunk 1 above)
Comment (joannakleong) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/azureFirewall.json]: Spoke with Yair, we will use the following description: The Firewall Internal Load Balancer IP to be used as the next hop in User Defined Routes

File: specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/azureFirewall.json [hunk 3]
```diff
@@ -0,0 +1,648 @@
...
+         "description":"Response for ListAzureFirewalls API service call."
+      },
+      "AzureFirewallApplicationRuleCollectionPropertiesFormat":{  
+         "properties":{  
+            "priority":{  
+               "type":"integer",
+               "format":"int32",
+               "maximum":65000,
+               "exclusiveMaximum":false,
+               "minimum":0,
```
[the remark this answers is not in the document]
Comment (joannakleong) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/azureFirewall.json]: In NRP looks like user can start from 100. Will update the minimum to reflect that. Thanks for catching

**Comment by joannakleong** [comment]
@sergey-shandar Pushed a final commit. Can this swagger be merged next Tuesday?

**Comment by dsgouda** [comment]
@sergey-shandar This good to merge. The dependent PR [here](https://github.com/Azure/azure-rest-api-specs/pull/3261) has been merged
Please consult with @lmazuel regarding the CI failures.

**Comment by sergey-shandar** [comment]
I spoke with @lmazuel, it's a timeout issue.

**Comment by joannakleong** [comment]
@sergey-shandar yes, this is because for private preview we named ourselves "Secure Gateway" and now we've finalized the name "Azure Firewall"

**Pull request merged**